src/main/java/mate/academy/hwspringbootintro/repository/spec/providers/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,23 @@
+package mate.academy.hwspringbootintro.repository.spec.providers;
+
+import mate.academy.hwspringbootintro.model.Book;
+import mate.academy.hwspringbootintro.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String AUTHOR_PROVIDER_KEY = "author";
+
+    @Override
+    public String getKey() {
+        return AUTHOR_PROVIDER_KEY;
+    }
+
+    @Override
+    public Specification<Book> getSpecification(String params) {
+
+        return (root, query, criteriaBuilder) ->
+            root.get(AUTHOR_PROVIDER_KEY).in(criteriaBuilder.literal(params.trim()));
```

Comment:
```suggestion
            root.get(AUTHOR_PROVIDER_KEY).in(criteriaBuilder.literal(params));
```
remove it in the other places.